### DISCUSSION TITLE
Memory (RAM) requirement for a 3.1GB genome indexing and alignment... ?

### DISCUSSION THREAD

**Discussion by DrIsh16** [alexdobin/STAR#1247] [Q&A]
I have human RNA-seq data that I need to analyze using human genomic reference of size 3.1GB.... I am using 16GB RAM with 256GB SSD system. The document associated with recent release (STAR 2.7.9a) does mentions that 'Mammal genomes require at least 16GB of RAM, ideally 32GB".

I started with genomeGenerate STAR runMode for the same with 1 thread 
command:
STAR --runThreadN 1 --runMode genomeGenerate --genomeDir . --genomeFastaFiles h38_genomic.fna --sjdbOverhang 99 --sjdbGTFfile h38_genomic.gtf

It seems to be stuck at "sorting Suffix Array chunks and saving them to disk" since long. Just need to make sure if the system configurations I am using are enough or I need more RAM?

**Comment by alexdobin** [comment]
Hi Isha,

to fit the genome into 16GB RAM, please use ```--genomeSAsparseD 3 --genomeSAindexNbases 12 --limitGenomeGenerateRAM 15000000000```

Cheers
Alex

**Reply by DrIsh16** [reply]
Thanks Alexander,
That's really kind of you to help.

Regards 
Isha
